lib/logproto/logproto-buffered-server.c

```diff
@@ -970,5 +961,5 @@ log_proto_buffered_server_init(LogProtoBufferedServer *self, LogTransport *trans
   else
     self->convert = (GIConv) -1;
   self->stream_based = TRUE;
-  self->pos_tracking = options->position_tracking_enabled;
+  self-> pos_tracking = (options->ack_tracker_type != ACK_INSTANT_BOOKMARKLESS);
```

Comment:
This code does not really help to extend ack trackers, as it is going to bite.
I was thinking about the following alternatives: (in my preference order)

1) The logsource is the one who pulls the strings here, and as such it should decide on which ack tracker should be used. The logproto could keep reporting if it is position tracked or not.

2) A dummy alternative (sure this not really solves the issue just makes it easier to detect if happens)
```c
gboolean _is_position_tracked(AckTrackerType act)
{
   switch (act)
   {
     case ACK_CONSECUTIVE:
        return TRUE;
     case ACK_INSTANT_BOOKMARKLESS:
        return FALSE;
    default:
       g_assert_not_reached();
   }

  g_assert_not_reached();
  return FALSE;
}
```

3) On the other hand it can be 'argued' that actually there is no other bookmarkless ack tracker implementation that is possible. So in the end this cover everything. (no change)

Reply:
@Kokan:
1:
The main difference here from previous versions is that the direction is _inverted_ between the type of `AckTracker` and the `pos_tracking` flag : `AckTrackerType` does not depend on position tracking but position tracking depends on the type of `AckTracker`.

2:
The referred `pos_tracking` is used by protos to decide whether they need to save position or not.
And the code expresses the fact that when ack tracker type is _not_ bookmarkless, then position is saved...

3:
Bookmarkless is special and we won't have another bookmarkless variants.